R-packages/covidcast/R/utils.R

```diff
@@ -237,6 +237,85 @@ abbr_to_name = function(abbr, ignore.case = FALSE, perl = FALSE, fixed = FALSE,
               ties_method = ties_method)
 }
 
+#' Get FIPS codes from state abbreviations 
+#'
+#' Look up FIPS codes by state abbreviations (including District of Columbia and
+#' Puerto Rico); this function is based on `grep()`, and hence allows for
+#' regular expressions.  
+#'
+#' @param abbr Vector of state abbreviations to look up.
+#' @param ignore.case,perl,fixed Arguments to pass to `grep()`, with the same
+#'   defaults as in the latter function. Hence, by default, regular expressions
```

Comment:
I think you mean does it make sense to use `ignore.case = TRUE`?  If so, yes, I can change it.

Also, it could make sense to use it throughout: I was just abiding by the `grep()` defaults with all these parameter choices.

Reply:
Oh yes I remember why I made `ignore.case = FALSE` the default, in addition to matching `grep()`'s default.  If you set `ignore.case = TRUE` and `fixed = TRUE`, then (annoyingly) `grep()` throws an error.  

So if `ignore.case = TRUE` is the default, then a user of our geo look up functions can't simply set `fixed = TRUE` in order to do non-regex matching, they'd have to both set `ignore.case = FALSE` and `fixed = TRUE`.

Reply:
Actually, I lied, `grep()` throws a warning (not an error).  So I think it's fine to set `ignore.case = TRUE` for the state abbr functions, if not throughout.

Reply:
Ok, my last commit sets `ignore.case = TRUE` just for state abbreviations.  Kept it the same beyond that.